### DISCUSSION TITLE
Does your libraries have possibility to call from C# via Marshalling?

### DISCUSSION THREAD

**Discussion by Rombersoft** [ImageMagick/ImageMagick#7093] [Development]
### ImageMagick version

7.1

### Operating system, version and so on

Debian 12

### Description

I need to convert images without reading and writing files from/to disk. Is the a possibility of working with stream if input stream is passed from the dotnet code and the output stream is returned to the dotnet code?

**Comment by dlemstra** [comment]
You can use this library: https://github.com/dlemstra/Magick.NET.

**Reply by Rombersoft** [reply]
I was there yesterday because I used their library but it has a problem with quality. And I've just seen that you commented on me there. https://github.com/dlemstra/Magick.NET/issues/741#issuecomment-1932829652
The ImageMagic native application on Linux OS makes better quality than the dotnet library that you advised me.
